src/common/Database/DatabaseWorkerPool.h

```diff
@@ -469,6 +476,102 @@ class DatabaseWorkerPool
             return _connectionInfo.database.c_str();
         }
 
+        bool ExecuteFile(char const* file)
+        {
+            T* t = GetFreeConnection();
+
+            auto handle = t->GetHandle();
+            if (!handle)
+                return false;
+
+            if (mysql_set_server_option(handle, MYSQL_OPTION_MULTI_STATEMENTS_ON))
+            {
+                sLog->outErrorDb("Cannot turn multi-statements on: %s", mysql_error(handle));
+                return false;
+            }
+
+            mysql_autocommit(handle, 0);
+            if (mysql_real_query(handle, "START TRANSACTION", sizeof("START TRANSACTION") - 1))
+            {
+                sLog->outErrorDb("Couldn't start transaction for db update file %s", file);
+                return false;
+            }
+
+            bool in_trasaction = true;
+            bool success = false;
+
+            if (FILE* fp = ACE_OS::fopen(file, "rb"))
+            {
+#if PLATFORM == PLATFORM_UNIX
+                flock(fileno(fp), LOCK_SH);
+#endif
+                //------
+
+                struct stat info;
+                fstat(fileno(fp), &info);
+
+                // if less than 1MB allocate on stack, else on heap
+                char* contents = (info.st_size > 1024 * 1024) ? new char[info.st_size] : (char*)alloca(info.st_size);
```

Comment:
Maybe the problem is here. It fills all the contents's size with thrash, and then fread replace it, but not all of it. [You can see it here](https://media.discordapp.net/attachments/262884535694393344/552318868542980126/unknown.png?width=1154&height=702)

Reply:
maybe use `sizeof(char)` as size and `info.st_size/sizeof(char)` as count for fread. Then the return value will tell you how many characters were read and you can use that as the parameter to mysql_real_query instead of `info.st_size`